micronaut-maven-plugin/src/test/java/io/micronaut/maven/testresources/StartTestResourcesServerMojoTest.java

```diff
@@ -0,0 +1,128 @@
+package io.micronaut.maven.testresources;
+
+import io.micronaut.maven.services.DependencyResolutionService;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.toolchain.ToolchainManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Unit test for StartTestResourcesServerMojo to verify skipTests functionality.
+ */
+class StartTestResourcesServerMojoTest {
+
+    @Mock
+    private MavenProject mavenProject;
+
+    @Mock
+    private MavenSession mavenSession;
+
+    @Mock
+    private DependencyResolutionService dependencyResolutionService;
+
+    @Mock
+    private ToolchainManager toolchainManager;
+
+    @Mock
+    private Plugin plugin;
+
+    private StartTestResourcesServerMojo mojo;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        mojo = new StartTestResourcesServerMojo(mavenProject, mavenSession, dependencyResolutionService, toolchainManager);
+        
+        when(mavenProject.getPlugin("io.micronaut.maven:micronaut-maven-plugin")).thenReturn(plugin);
+        when(mavenSession.getUserProperties()).thenReturn(new Properties());
+        when(mavenSession.getSystemProperties()).thenReturn(new Properties());
+    }
+
+    @Test
+    void shouldSkipWhenSkipTestsIsTrue() {
+        // Given
+        Properties userProps = new Properties();
+        userProps.setProperty("skipTests", "true");
+        when(mavenSession.getUserProperties()).thenReturn(userProps);
+
+        // When
+        boolean result = mojo.isTestExecutionSkipped();
+
+        // Then
+        assertTrue(result, "Should skip when skipTests is true");
+    }
+
+    @Test
+    void shouldSkipWhenMavenTestSkipIsTrue() {
+        // Given
+        Properties userProps = new Properties();
+        userProps.setProperty("maven.test.skip", "true");
+        when(mavenSession.getUserProperties()).thenReturn(userProps);
+
+        // When
+        boolean result = mojo.isTestExecutionSkipped();
+
+        // Then
+        assertTrue(result, "Should skip when maven.test.skip is true");
+    }
+
+    @Test
+    void shouldSkipWhenSkipITsIsTrue() {
+        // Given
+        Properties userProps = new Properties();
+        userProps.setProperty("skipITs", "true");
+        when(mavenSession.getUserProperties()).thenReturn(userProps);
+
+        // When
+        boolean result = mojo.isTestExecutionSkipped();
+
+        // Then
+        assertTrue(result, "Should skip when skipITs is true");
+    }
```

Comment:
Use a parameterised test, since they are all essentially the same with just different property